Makefile

```diff
@@ -17,154 +17,163 @@
 # * limitations under the License.
 # *
 
-TARGET_EXEC ?= hal_test
-
+# Terminal colors
 RED='\033[0;31m'
 GREEN='\033[0;32m'
 YELLOW='\033[0;33m'
 NC='\033[0m'
 ECHOE = /bin/echo -e
+MKDIR_P ?= @mkdir -p
 
-$(info $(shell ${ECHOE} ${GREEN}TARGET_EXEC [$(TARGET_EXEC)]${NC}))
-
-UT_CORE_DIR :=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
-export PATH := $(shell pwd)/toolchain:$(PATH)
+# Directories and PATH setup
+UT_CORE_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+export PATH := $(UT_CORE_DIR)/toolchain:$(PATH)
 
-# defaults for target arm
-ifeq ($(TARGET),arm)
-CUNIT_VARIANT=arm-rdk-linux-gnueabi
-#CC := arm-rdk-linux-gnueabi-gcc -mthumb -mfpu=vfp -mcpu=cortex-a9 -mfloat-abi=soft -mabi=aapcs-linux -mno-thumb-interwork -ffixed-r8 -fomit-frame-pointer
-# CFLAGS will be overriden by Caller as required
-INC_DIRS += $(UT_CORE_DIR)/sysroot/usr/include
-TARGET = arm
-else
-TARGET = linux
-endif
+# Set target based on TARGET variable
+TARGET := $(if $(TARGET),$(TARGET),linux)
 export TARGET
 
-$(info TARGET [$(TARGET)])
-
 # Moveable Directories based on the caller makefile
 TOP_DIR ?= $(UT_CORE_DIR)
 BIN_DIR ?= $(TOP_DIR)/build/bin
-LIB_DIR ?= $(TOP_DIR)/build/${TARGET}/lib
+LIB_DIR ?= $(TOP_DIR)/build/$(TARGET)/lib
 BUILD_DIR ?= $(TOP_DIR)/build/$(TARGET)/obj
 
 # Non-Moveable Directories
 FRAMEWORK_DIR = $(UT_CORE_DIR)/framework
 UT_CONTROL = $(FRAMEWORK_DIR)/ut-control
 
+# Flags
 XCFLAGS := $(KCFLAGS)
+INC_DIRS += $(UT_CORE_DIR)/include $(UT_CONTROL)/include $(UT_CORE_DIR)/src
```

Comment:
we really should be having a `${FRAMEWORK}/shared_include` rather than `${UT_CONTROL}/include` so you can add more libraries in the future, and ut-core knows where the header files are.

Then you use it in the code like this, since ${FRAMEWORK} is in the path.

```c
include <ut-control/ut-control-blah.h>
````

But we can do this later.